packages/components/psammead-podcast-promo/src/index.test.jsx

```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
+import * as scripts from '@bbc/gel-foundations/scripts';
+
+import BasicExample from './examples/basic';
+
+describe('Podcast Promo', () => {
```

Comment:
could we have unit tests for some of the acceptance criteria especially the a11y stuff?

Reply:
Only a11y thing implemented on the psammead side is aria-hiding the call-to-action - added test for that, but I'm not even sure we should be aria-hiding content in psammead

I know one snapshot test looks really barren here, but there's just no business logic in this component (not even a conditional) and I think snapshots are the best way of testing CSS and HTML attributes

Reply:
should we test that the component meets the acceptance criteria? e.g. renders the title, renders the description, renders the image, image has alt text. these things can break and would be hard to spot with one snapshot. I know there's no logic and we have 100% coverage with just 1 snapshot but it's good to have the confidence a component meets it's requirements and will be protected against any regressions with specific unit tests

Reply:
There's not one component here though - there are 9, with the idea being that they can be composed together (and with other components and HTML) to implement a range of criteria.

The acceptance criteria on the ticket are specific to how we want it used in Simorgh.  The example in `/examples/src` show how to use the component in simorgh to meet the simorgh AC, but we shouldn't assume it is always going to be used that way.

If we're creating a Psammead component, it is supposed to be a generic, reusable library.

eg, we shouldn't be forcing other psammead users to have stuff like `a section with a role of region` or `screenreaders to read out "[Brand Title], Episodes"` if they don't want to.  Those are our goals - not necessarily the goals of other psammead users

I know those "other users" barely exist, but if we want to start to solve that, we shouldn't be filling psammead with inflexible components hardcoded to Simorgh's use case

Alternatively, if we are creating a component we only want to be relevant to Simorgh, I think we should just create it in Simorgh - it makes our lives a lot easier.

Reply:
That said, I do agree the snapshot is too noisy though - I can test each of the 9 components individually

But those tests would just be things  like `a card-title is an h3 with these css attributes` and I think tests like those are best represented as snapshots anyway - curious what others think

Reply:
I'm curious to hear others think too. I'm not totally convinced that snapshots alone are good quality tests. Any change made will cause the tests to fail. They don't point out regressions, just that something has changed. It's too easy to accept the change and update the snapshots and then potentially lose original acceptance criteria.

Reply:
I think we should test that the elements specified by  the a11y requirements are met. e.g. title should be an h3 element. But the css attributes shouldn't matter since they are not part of the requirements. I don't think specifying elements should be part of a11y requirements but I don't know enough about a11y to know if that is good or bad practice.

I do think that snapshots can miss this requirements quite easily and we need better tests to ensure that when the requirements change, someone needs to go in the tests to change them, without simply updating snapshots